### ISSUE TITLE
No sane way to change user’s profile picture in GNOME

### ISSUE THREAD

**Issue by michalrus** [NixOS/nixpkgs#20872]
## Issue description

In GNOME, each regular user can have a picture associated with their account. This picture will be shown e.g. in the GDM menu.

Currently, the only way to change the picture is to edit `/var/lib/AccountsService/users/${user.name}` and add a line similar to:

```
Icon=/run/current-system/sw/share/pixmaps/faces/soccerball.png
```

### Steps to reproduce

Install GNOME → open System Settings → Users → click on the default user picture → try to change it to something else. Nothing happens.

Corresponding `journalctl` entry:

```
Dec 02 23:35:19 nixos org.gnome.Shell.desktop[2689]: (gnome-control-center:3078):
   AccountsService-WARNING **: SetIconFile call failed:
   GDBus.Error:org.freedesktop.Accounts.Error.PermissionDenied:
   Authentication is required
```

## Technical details

* System: 16.09.1135.b833b10 (Flounder)
* Nix version: nix-env (Nix) 1.11.4
* Nixpkgs version: "16.09.1135.b833b10"

**Comment by chris-martin** [comment]
It would be nice to be able to specify a picture via the NixOS config. Would that be incompatible with fixing this GUI?

**Comment by michalrus** [comment]
Well, it would seem better to do it in `configuration.nix`, right. But on the other hand, profile picture configuration files are in `/var/lib/AccountsService/users/`, so in a mutable location. Perhaps, apart from this mutable location, we could take UX into consideration… One might change profile pictures relatively often.

Also, I think this location is used by lightdm, too?

**Comment by jtojnar** [comment]
Duplicate of #10025

**Comment by michalrus** [comment]
@jtojnar, @grahamc, really? `/var/lib/AccountsService/users/${user.name}` sounds quite mutable to me.

**Comment by michalrus** [comment]
More like sub-issue. (ﾉ^_^)ﾉ

**Comment by jtojnar** [comment]
Either way, changing the image from the Control Center works for me on GNOME 3.24 except for setting a custom image, which will be fixed by #29983.
